code/modules/virology/virology_goals.dm

```diff
@@ -0,0 +1,205 @@
+
+/datum/virology_goal/virus/New()
+	var/first_loop = TRUE
+	while(check_for_duplicate() || first_loop)
+		first_loop = FALSE
+		goal_symptoms = list()
+		var/list/datum/symptom/symptoms = subtypesof(/datum/symptom)
+		var/stealth = 0
+		for(var/i in 1 to 5)
+			var/list/datum/symptom/candidates = list()
+			for(var/datum/symptom/V as anything in symptoms) //There is a "as anything" added because for some mystery reason it doesnt work without it
+				var/datum/symptom/S = V
+				if(stealth + S.stealth >= 3) //The Pandemic cant detect a virus with stealth 3 or higher and we dont want that, this isnt a stealth virus
+					continue
+				candidates += S
+			var/datum/symptom/S2 = pick(candidates)
+			goal_symptoms += S2
+			stealth += S2.stealth
+			symptoms -= S2
+
+/datum/virology_goal/virus/check_for_duplicate()
+	. = FALSE
+	if(!length(goal_symptoms))
+		return
+	var/goals = GLOB.archived_virology_goals + GLOB.virology_goals
+	for(var/datum/virology_goal/virus/V in goals)
+		if(goal_symptoms == V.goal_symptoms)
+			return TRUE
+
+/datum/virology_goal/virus/get_report()
+	return {"<b>Specific Viral Sample Request (Non-Stealth)</b><br>
+	A specific viral sample is required for confidential reasons. We need you to deliver [delivery_goal]u of viral samples with exactly only the following symptoms: [symptoms_list2text()] to us through the cargo shuttle.
+	<br>
+	-Nanotrasen Virology Research"}
+
+/datum/virology_goal/virus/get_ui_report()
+	return {"A specific viral sample is required for confidential reasons. We need you to deliver [delivery_goal]u of viral samples with exactly only the following symptoms: [symptoms_list2text()] to us through the cargo shuttle."}
+
+/datum/virology_goal/virus/proc/symptoms_list2text()
+	var/list/msg = list()
+	for(var/datum/symptom/S as anything in goal_symptoms)
+		msg += initial(S.name)
+	return jointext(msg, ", ")
+
+/datum/virology_goal/virus/check_completion(list/datum/reagent/reagent_list)
+	. = FALSE
+	var/datum/reagent/blood/BL = locate() in reagent_list
+	if(BL && BL.data && BL.data["viruses"])
+		for(var/datum/disease/advance/D in BL.data["viruses"])
+			if(length(D.symptoms) != length(goal_symptoms)) //This is here so viruses with extra symptoms dont get approved
+				return
+			for(var/S in goal_symptoms)
+				var/datum/symptom/SY = locate(S) in D.symptoms
+				if(!SY)
+					return
+				delivered_amount = clamp(delivered_amount + BL.volume, 0, delivery_goal)
+				if(delivered_amount >= delivery_goal)
+					completed = TRUE
+					check_total_virology_goals_completion()
+					return TRUE
+
+/datum/virology_goal/virus/stealth
+	name = "Specific Viral Sample Request (Stealth)"
+
+/datum/virology_goal/virus/stealth/New()
+	var/first_loop = TRUE
+	while(check_for_duplicate() || first_loop)
+		first_loop = FALSE
+		goal_symptoms = list()
+		var/list/datum/symptom/symptoms = subtypesof(/datum/symptom)
+		var/stealth = 0
+		for(var/i in 1 to 5)
+			var/list/datum/symptom/candidates = list()
```

Comment:
I don't think dm *really* supports typed lists, so this might be why your 'as anything' was needed here. With this change in place, you probably won't need it.
```suggestion
		var/list/symptoms = subtypesof(/datum/symptom)
		var/stealth = 0
		for(var/i in 1 to 5)
			var/list/candidates = list()
```

Reply:
merged the requested change but still doesnt work without the "as anything", gonna leave it as is